src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -295,6 +315,24 @@ void ClusterTaskManager::DispatchScheduledTasksToWorkers(
         continue;
       }
 
+      if (sched_cls_info.running_tasks.size() >= sched_cls_info.capacity) {
+        if (get_time_() > sched_cls_info.next_update_time) {
+          // Don't increase the capacity on the very first update since that
+          // will unblock a new task immediately, instead of after the timeout.
+          if (sched_cls_info.num_updates > 0) {
+            sched_cls_info.capacity++;
+          }
+
+          double wait_time =
+              (1e6 * sched_cls_cap_interval_ms_) * (1L << sched_cls_info.num_updates++);
+          sched_cls_info.next_update_time = get_time_() + wait_time;
+          sched_cls_info.timer =
```

Comment:
Do we need the new timer? I thought that ScheduleAndDispatchTasks is already called periodically from the node manager. If it's not, can we move the timer there instead?

Reply:
Agree we should try to avoid extra timers. If we ensure the regular scheduling is <= 1s, I think we can just skip this. It's not necessary for the delay to be precise.